server/src/test/java/org/opensearch/rest/action/admin/cluster/RestNodesStatsActionTests.java

```diff
@@ -134,6 +134,8 @@ public void testIndexMetricsRequestWithoutIndicesMetric() throws IOException {
         final HashMap<String, String> params = new HashMap<>();
         final Set<String> metrics = new HashSet<>(RestNodesStatsAction.METRICS.keySet());
         metrics.remove("indices");
+        // caches stats is handled separately
+        metrics.remove("caches");
```

Comment:
Probably needs method to be renamed `testIndexMetricsRequestWithoutIndicesMetric` -> `testIndexMetricsRequestWithoutIndicesAndCachesMetric`

Reply:
This test targets for the case that if `index_metric` is requested, the parent level `indices` metric must also be requested, if not an exception is thrown, if we change the method name to `testIndexMetricsRequestWithoutIndicesAndCachesMetrics` may introduce confusion, it seems to test that if `index_metric` is requested, both the parent level `indices` and `caches` metrics must be requested.

Reply:
This is super minor, I was just under impression that ` metrics.remove("indices");` corresponds to `WithoutIndicesMetric`, but since we also do `metrics.remove("caches");`, it looks logical to reflect that as well, no?

Reply:
Sure, I've changed that, thank you!